### ISSUE TITLE
write the aimmo setup script

### ISSUE THREAD

**Issue by dent50cent** [ocadotechnology/aimmo#819]
### Description
Create the setup script for aimmo, this issue is only concerned with the setup for mac, further issues will add to the script for other operating systems.

#### Requirements
- [x] The script should run for mac
- [x] It should simply the process of setting up aimmo, setting up completely ready for use if possible.

**Comment by OlafSzmidt** [comment]
Duplicate of #817? Plus I don't think these setup scripts are actually that great of an idea. Ubuntu for e.g. is different version to version. I can't follow the same instructions for minikube that in 18 and 16 at the same time.

Stuff like removing cmdtest to install yarn should also be done by the user imo

**Comment by dent50cent** [comment]
I copied over the description since they're relating to the same epic until I can come up with a better description. Plus, this idea was discussed in the latest sprint planning session and was decided by the team that it would be worth looking into. The reason for attempting this was a quality of life improvement for new starters or people who for whatever reason needed to restart their work on the project and needed to set up the dependencies again. 

The rest of your comment is a little hard to follow.

**Comment by OlafSzmidt** [comment]
Sorry, wrote it while commuting 😅 Having set up the project recently myself on Ubuntu, using a setup script was more trouble than just following docs.

There's three issues I have with setup scripts;

1. All the edge cases, not only between OS's but also between versions are a pain. Each Linux behaves differently.
2. We don't want to remove stuff and add stuff on behalf of the user globally on their computers. Ubuntu 16+ requires cmdtest to be removed, what if the user requires that package and we are removing it?
3. Maintainability, stuff will change all the time and we have to maintain it for each OS. 

I think the repo is missing good docs rather than setup scripts. It was easier for me to set it up using Google rather than running ubuntu_setup.sh

**Comment by dent50cent** [comment]
Ah okay 😅 I understand what you mean more now. So to solve the issue between operating systems. There are various python libraries that allow us to detect a user's operating system and make subprocess calls, such as on the command line, so we can just write out the process per OS within these subprocess calls. This also gives the benefit of consistent error checking between the scripts across these operating systems. 

As for the cmdtest issue, a simple prompt could allow us to ask the user if removing it is okay. If not then we wouldn't remove it, but the user then wouldn't be able to work on the project.

Finally the maintainability, this again comes back to the idea of using python, since we store the versions of the software we are using, in theory, this script can be extended to find the versions of stuff that we require. If not, it would simply need to be updated along with the readme page which would add no more than a few minutes to the task. As a final part to this task, I would update and hopefully improve the docs to reflect these things.

**Comment by dent50cent** [comment]
Contains Duplicate code from [#823](https://github.com/ocadotechnology/aimmo/issues/823). So this issue can be closed.
